pype/plugins/global/publish/integrate_new.py

```diff
@@ -952,21 +954,37 @@ def handle_destination_files(self, integrated_file_sizes, mode):
         """
         if integrated_file_sizes:
             for file_url, _file_size in integrated_file_sizes.items():
+                if not os.path.exists(file_url):
+                    self.log.debug(
+                        "File {} was not found.".format(file_url)
+                    )
+                    continue
+
                 try:
                     if mode == 'remove':
-                        self.log.debug("Removing file ...{}".format(file_url))
+                        self.log.debug("Removing file {}".format(file_url))
                         os.remove(file_url)
                     if mode == 'finalize':
-                        self.log.debug("Renaming file ...{}".format(file_url))
-                        import re
-                        os.rename(file_url,
-                                  re.sub('\.{}$'.format(self.TMP_FILE_EXT),
-                                         '',
-                                         file_url)
-                                  )
-
-                except FileNotFoundError:
-                    pass  # file not there, nothing to delete
+                        new_name = re.sub(
+                            r'\.{}$'.format(self.TMP_FILE_EXT),
+                            '',
+                            file_url
+                        )
+
+                        if os.path.exists(new_name):
```

Comment:
Is possible that `new_name == file_url`? In that case I would skip.

Reply:
No, `new_name` is `file_url + self.TMP_FILE_EXT`.